### PR TITLE
Fix: handle multiple release pages when listing all versions

### PR DESCRIPTION
**Pull request by dhanak**
Also fix behavior with GITHUB_API_TOKEN envvar (repo URL was missing).

### CODE REVIEW

File: bin/list-all [hunk 1]
```diff
@@ -5,15 +5,23 @@ function sort_versions() {
     LC_ALL=C sort -gt. -k 1,1 -k 2,2n -k 3,3n -k 4,4n -k 5,5n | sort -r - | awk '{print $2}'
 }
 
-cmd="curl -Ls0"
-releases_path="https://api.github.com/repos/JuliaLang/julia/releases"
+releases=$(tempfile -s .json)
+headers=$(tempfile)
```
Comment (lassepe) [inline bin/list-all:9]: `tempfile` is not available on non-debian-based linux distributions. Thus, this PR broke asdf-julia for Arch Linux, Alpine et al.

Comment (dhanak) [reply bin/list-all:9]: Ah, pity. I guess `mktemp` would do the trick then, wouldn't it? I'll just replace it with that, then.

Comment (lassepe) [reply bin/list-all:9]: I already did that [here](https://github.com/rkyleg/asdf-julia/pull/26). But if you could test whether that PR works on your machine that would be helpful. I don't have access to Ubuntu right now.

Comment (dhanak) [reply bin/list-all:9]: See here https://github.com/rkyleg/asdf-julia/pull/27.

Comment (dhanak) [reply bin/list-all:9]: :-) Ok then. I dropped my PR, fetched yours and tested it on my machine. Good news is, tt works just fine.

**Pull request merged**